proxy_connect.go

```diff
@@ -146,25 +155,18 @@ func addDialDeadlineIfNecessary(ctx context.Context, req *http.Request) (context
 func noopCancel() {
 }
 
-func respondOK(resp *http.Response, req *http.Request, ctx filters.Context) (*http.Response, filters.Context) {
```

Comment:
In the old code, you can examine references to `ctxKeyNoRespondOkay` to see that we only suppress the OK response if the connection comes in through `proxy.Connect` ([here](https://github.com/getlantern/proxy/blob/master/proxy_connect.go#L161)).  Rather than making this choice and tracking it as state on an object, it seems more straight-forward to pass the choice as a parameter through the call-stack.

Reply:
That's fine, but you could still keep lines 152-154 which are repeated.